src/hotspot/os/linux/cgroupSubsystem_linux.cpp

```diff
@@ -528,6 +528,8 @@ jlong CgroupSubsystem::memory_limit_in_bytes() {
   if (!memory_limit->should_check_metric()) {
     return memory_limit->value();
   }
+  jlong phys_mem = os::Linux::physical_memory();
+  log_trace(os, container)("total physical memory: " JLONG_FORMAT, phys_mem);
```

Comment:
Fair enough. You're right, it only works because the first match is prior to cgroups overwriting the physical number. I'm happy to wait until we merge #9880.